vpr/src/place/place.cpp

```diff
@@ -2636,177 +3203,525 @@ static void update_bb(ClusterNetId net_id, t_bb* bb_coord_new, t_bb* bb_edge_new
+            bb_coord_new.ymin = curr_bb_coord->ymin;
+            bb_edge_new.ymin = curr_bb_edge->ymin;
         }
         /* End of move down case. */
 
-    } else if (ynew > yold) { /* Moved up. */
+    } else if (pin_new_loc.y > pin_old_loc.y) { /* Moved up. */
 
         /* Update the ymin fields for coordinates and number of edges first. */
 
-        if (yold == curr_bb_coord->ymin) { /* Old position at ymin. */
+        if (pin_old_loc.y == curr_bb_coord->ymin) { /* Old position at ymin. */
             if (curr_bb_edge->ymin == 1) {
                 get_bb_from_scratch(net_id, bb_coord_new, bb_edge_new);
                 bb_updated_before[net_id] = GOT_FROM_SCRATCH;
                 return;
             } else {
-                bb_edge_new->ymin = curr_bb_edge->ymin - 1;
-                bb_coord_new->ymin = curr_bb_coord->ymin;
+                bb_edge_new.ymin = curr_bb_edge->ymin - 1;
+                bb_coord_new.ymin = curr_bb_coord->ymin;
             }
         } else { /* Moved up, old position was not at ymin. */
-            bb_coord_new->ymin = curr_bb_coord->ymin;
-            bb_edge_new->ymin = curr_bb_edge->ymin;
+            bb_coord_new.ymin = curr_bb_coord->ymin;
+            bb_edge_new.ymin = curr_bb_edge->ymin;
         }
 
         /* Now do the ymax fields for coordinates and number of edges. */
 
-        if (ynew > curr_bb_coord->ymax) { /* Moved past ymax. */
-            bb_coord_new->ymax = ynew;
-            bb_edge_new->ymax = 1;
-        } else if (ynew == curr_bb_coord->ymax) { /* Moved to ymax */
-            bb_coord_new->ymax = ynew;
-            bb_edge_new->ymax = curr_bb_edge->ymax + 1;
+        if (pin_new_loc.y > curr_bb_coord->ymax) { /* Moved past ymax. */
+            bb_coord_new.ymax = pin_new_loc.y;
+            bb_edge_new.ymax = 1;
+        } else if (pin_new_loc.y == curr_bb_coord->ymax) { /* Moved to ymax */
+            bb_coord_new.ymax = pin_new_loc.y;
+            bb_edge_new.ymax = curr_bb_edge->ymax + 1;
         } else { /* ymax unchanged. */
-            bb_coord_new->ymax = curr_bb_coord->ymax;
-            bb_edge_new->ymax = curr_bb_edge->ymax;
+            bb_coord_new.ymax = curr_bb_coord->ymax;
+            bb_edge_new.ymax = curr_bb_edge->ymax;
         }
         /* End of move up case. */
 
-    } else { /* ynew == yold -- no y motion. */
-        bb_coord_new->ymin = curr_bb_coord->ymin;
-        bb_coord_new->ymax = curr_bb_coord->ymax;
-        bb_edge_new->ymin = curr_bb_edge->ymin;
-        bb_edge_new->ymax = curr_bb_edge->ymax;
+    } else { /* pin_new_loc.y == yold -- no y motion. */
+        bb_coord_new.ymin = curr_bb_coord->ymin;
+        bb_coord_new.ymax = curr_bb_coord->ymax;
+        bb_edge_new.ymin = curr_bb_edge->ymin;
+        bb_edge_new.ymax = curr_bb_edge->ymax;
     }
 
     if (bb_updated_before[net_id] == NOT_UPDATED_YET) {
         bb_updated_before[net_id] = UPDATED_ONCE;
     }
 }
 
+static void update_layer_bb(ClusterNetId net_id,
```

Comment:
These new *layer* routines are under-commented; there isn't any overall comment talking about the per-layer approach vs. the 3D approach -- there should be one.

Reply:
I will have this overall comments in the next PR that I have done the refactoring